packages/backend/src/services/SpaceService/SpaceService.test.ts

```diff
@@ -0,0 +1,865 @@
+                    access: { spaceRole: SpaceMemberRole.VIEWER },
+                    action: 'update',
+                    expectedResult: false,
+                    contentType: 'Dashboard',
+                },
+                {
+                    name: 'cannot update dashboard when group space role is viewer',
+                    user: { projectRole: ProjectMemberRole.EDITOR },
+                    space: { isPrivate: true },
+                    access: { groupSpaceRole: SpaceMemberRole.VIEWER },
+                    action: 'update',
+                    expectedResult: false,
+                    contentType: 'Dashboard',
+                },
+
+                // Promoting
+                // This fails because editors have manage permission on dashboards, so they can do any action (including promote)
+                // The intended behaviour is that only developers can promote dashboards
+                // {
+                //     name: 'cannot promote dashboard in public space',
+                //     user: { projectRole: ProjectMemberRole.EDITOR },
+                //     space: { isPrivate: false },
+                //     access: {},
+                //     action: 'promote',
+                //     expectedResult: false,
+                //     contentType: 'Dashboard',
+                // },
+            ])(
+                '$name',
+                async ({
+                    user,
+                    space,
+                    access,
+                    action,
+                    expectedResult,
+                    contentType,
+                }) => {
+                    const testUser = createTestUser(user);
+                    const testSpace = createTestSpace(space);
+                    const response = createSpaceAccessResponse({
+                        ...user,
+                        ...access,
+                        isPrivate: space.isPrivate,
+                    });
+
+                    tracker.on.select('spaces').response([response]);
+
+                    const result = await service._userCanActionSpace(
+                        testUser,
+                        contentType as 'Space' | 'Dashboard' | 'Chart',
+                        testSpace,
+                        action as AbilityAction,
+                    );
+                    try {
+                        expect(result).toBe(expectedResult);
+                    } catch (error) {
+                        await service._userCanActionSpace(
+                            testUser,
+                            contentType as 'Space' | 'Dashboard' | 'Chart',
+                            testSpace,
+                            action as AbilityAction,
+                            true,
+                        );
+                        throw error;
+                    }
+                },
+            );
+        });
+
+        describe('project developers', () => {
+            it.each([
```

Comment:
we could add more scenarios like `cannot promote dashboard in private space` or other things admins can do but developers cannot